t/porting/deprecation.t

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/perl
+
+BEGIN {
+  if (-f './TestInit.pm') {
+    @INC = '.';
+  } elsif (-f '../TestInit.pm') {
+    @INC = '..';
+  }
+}
+use TestInit qw(T); # T is chdir to the top level
+
+use warnings;
+use strict;
+use Config;
+use Data::Dumper;
+require './t/test.pl';
+
+plan("no_plan");
+
+# Test that all deprecations in regen/warnings.pl are mentioned in
+# pod/perldeprecation.pod and that there is sufficient time between them.
+
+my $pod_file = "./pod/perldeprecation.pod";
+my $warnings_file = "./regen/warnings.pl";
+
+do $warnings_file;
+our $WARNING_TREE;
+
+my $deprecated = $WARNING_TREE->{all}[1]{deprecated}[2];
+
+open my $fh, "<", $pod_file
+    or die "failed to open '$pod_file': $!";
+my $removed_in_version;
+my $subject;
+my %category_seen;
+my %subject_has_category;
+my $in_legacy;
+
+while (<$fh>) {
+    if (/=head2 (?|Perl (5\.\d+)(?:\.\d+)?|(Unscheduled))/) { # ignore minor version
+        $removed_in_version = lc $1;
+        if ($removed_in_version eq "5.38") {
+            $in_legacy = 1;
+        }
+    }
+    if (/=head3 (.*)/) {
```

Comment:
And also maybe `elsif`, since we don't expect any one line to match more than one of them

Reply:
sure

Reply:
Done.